### ISSUE TITLE
Autostart for all users

### ISSUE THREAD

**Issue by selaquik** [Rem0o/FanControl.Releases#234]
Hello 

![image](https://user-images.githubusercontent.com/83961515/117662978-81e6a100-b1e3-11eb-8bd6-827088c2dccc.png)

I tried using the autostart setting - but it didn't work.

I managed to solve this problem by adding the program to the autostart of the windows. However, it does not work for other users.

Can there be a version with installation? Or how to make autostart on all users? Thank you

**Comment by Rem0o** [comment]
Make sure its location is in a folder where every user has access to. Common way to do it is in "Program Files".

**Comment by selaquik** [comment]
Yes, I put the program files in the folder and gave full access. But it doesn't work.

**Comment by Rem0o** [comment]
Anything in your log.txt?

**Comment by selaquik** [comment]
[log.txt](https://github.com/Rem0o/FanControl.Releases/files/6507836/log.txt)


I tried to create a separate second account and fancontrol folder, but I ran into a problem that the fans cannot see
![Screenshot_1](https://user-images.githubusercontent.com/83961515/118797081-8dbb1d00-b8df-11eb-8abb-61c9e7659c1e.png)

**Comment by Rem0o** [comment]
That's the driver failing to load in the background. Either you have 2 instances running, or you tried to start it too soon.

**Comment by Rem0o** [comment]
From your log "NVAPI_INVALID_USER_PRIVILEGE".

User must be admin to run the tool.

**Comment by davmin** [comment]
My daily driver user account does not have admin privileges, I have a secondary account for that. I assume that's why my auto start task was also failing. Despite having it checked, I would have to run it as admin and enter my admin credentials in the UAC.

I went in to the task scheduler and made a copy of the task that runs at startup instead of login, and using the SYSTEM user. This works but has the downside of fancontrol not being accessible via the system tray.

**Comment by Rem0o** [comment]
>  This works but has the downside of fancontrol not being accessible via the system tray.

Well, from a non-admin account, how would you expect to have control over an admin-level application?

**Comment by davmin** [comment]
I had no expectation that launching it as SYSTEM prior to any user login would let me edit it in a user session. And I don't have a problem that it doesn't! If I want to make changes I can kill it via task manager and open it to make my changes, or just edit the json manually. But I felt it's probably worth mentioning the drawbacks of my solution when suggesting a workaround to other users. 

For what its worth, when elevating with an admin user at the UAC prompt, while logged into an non-admin account, the behavior is that the application _does_ launch and show up in the system tray of the non-admin user and the non-admin user can edit settings etc. My only issue was that it was not auto starting.

Ideally though I would like a background service and a user client application. User sets a setting in the client application, the service makes the changes with SYSTEM authority. But I'm happy with the application as is and will continue to use it. Even with this workaround its better than running some bloated mobo manufacturer's mess. Thanks for the hard work!

**Comment by IslandJohn** [comment]
You could have a background process (e.g. running at SYSTEM) that runs the fan control side and the UI from the limited access user would be through some IPC. The background process could accept chose configuration and run with it.

**Comment by Rem0o** [comment]
https://github.com/Rem0o/FanControl.Releases/issues/135
